src/pages/OriginalSampleRegistration.tsx

```diff
@@ -146,15 +115,124 @@ function OriginalSampleRegistration({ registrationInfo }: RegistrationParams) {
+          ...memo,
+          ...tissue.blocks.map<OriginalSampleData>((block) => {
+            const sampleRegisterData: OriginalSampleData = {
+              species: tissue.species.trim(),
+              donorIdentifier: tissue.donorId.trim(),
+              externalIdentifier: block.externalIdentifier ? block.externalIdentifier.trim() : undefined,
+              hmdmc: tissue.hmdmc.trim(),
+              labwareType: block.labwareType.trim(),
+              lifeStage: tissue.lifeStage,
+              tissueType: tissue.tissueType.trim(),
+              spatialLocation: block.spatialLocation,
+              replicateNumber: block.replicateNumber ?? undefined,
+              fixative: block.fixative.trim(),
+              solution: block.solution.trim(),
+              sampleCollectionDate: tissue.sampleCollectionDate
+                ? tissue.sampleCollectionDate instanceof Date
+                  ? tissue.sampleCollectionDate.toLocaleDateString()
+                  : tissue.sampleCollectionDate
+                : undefined
+            };
+
+            return sampleRegisterData;
+          })
+        ];
+      }, []);
+      setSampleRegistrationData(samples);
+      return new Promise((resolve) => {
+        resolve({ request: { samples } });
+      });
+    },
+    [setSampleRegistrationData]
+  );
+
+  const resultColumns: Array<Column<LabwareResultData>> = React.useMemo(() => {
+    return [
+      {
+        Header: 'Barcode',
+        id: 'barcode',
+        accessor: (result: LabwareResultData) => result.labware.barcode
+      },
+      {
+        Header: 'Labware Type',
+        id: 'labwareType',
+        accessor: (result: LabwareResultData) => result.labware.labwareType.name
+      },
+      {
+        Header: 'External ID',
+        id: 'externalID',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(result.labware, (sample: SampleFieldsFragment) => sample.tissue.externalName ?? '')
+      },
+      {
+        Header: 'Donor ID',
+        id: 'donorId',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(result.labware, (sample: SampleFieldsFragment) => sample.tissue.donor.donorName ?? '')
+      },
+      {
+        Header: 'Tissue type',
+        id: 'tissueType',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(
+            result.labware,
+            (sample: SampleFieldsFragment) => sample.tissue.spatialLocation.tissueType.name ?? ''
+          )
+      },
+      {
+        Header: 'Spatial location',
+        id: 'spatialLocation',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(result.labware, (sample: SampleFieldsFragment) => String(sample.tissue.spatialLocation.code))
+      },
+      {
+        Header: 'Replicate',
+        id: 'replicate',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(result.labware, (sample: SampleFieldsFragment) => String(sample.tissue.replicate))
```

Comment:
can sample.tissue.replicate be undefined here? does it need a `?? ""`

Reply:
Yes, replicate can be undefined in case of original sample registration